core/src/mast/node/basic_block_node/mod.rs

```diff
@@ -208,21 +210,19 @@ impl BasicBlockNode {
 /// Mutators
 impl BasicBlockNode {
     /// Sets the provided list of decorators to be executed before all existing decorators.
-    pub fn prepend_decorators(&mut self, decorator_ids: Vec<DecoratorId>) {
-        let mut new_decorators: DecoratorList =
-            decorator_ids.into_iter().map(|decorator_id| (0, decorator_id)).collect();
-        new_decorators.extend(mem::take(&mut self.decorators));
-
-        self.decorators = new_decorators;
+    pub fn prepend_decorators(&mut self, decorator_ids: DecoratorSpan) {
+        // At which point do we want to create a decorator span? Do we convert the vector into a single span?
```

Comment:
Here the calling function logic depends on whether the spans are contiguous, but in this case the assumption is that they are, and therefore `prepend_decorators` accepts a single `DecorartorSpan`.